### PR TITLE
Telink B91 PWM

### PR DESCRIPTION
**Pull request by yurvyn**

### CODE REVIEW

File: dts/bindings/pwm/telink,b91-pwm.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    pinctrl-0:
+      type: phandles
+      required: true
+
+    clock-frequency:
+      type: int
+      required: true
+      description: Default PWM Peripheral Clock frequency in Hz (is used if 32K Clock is disabled)
+
+    clk32k-ch0-enable:
```
Comment (mbolivar-nordic) [inline dts/bindings/pwm/telink,b91-pwm.yaml:22]: It seems like it might be more ergonomic to make these optional booleans. Did you consider that? If so, why did you choose a required int?

File: boards/riscv/tlsr9518adk80d/doc/index.rst [hunk 2]
```diff
@@ -1,4 +1,4 @@
-.. _tlst9518adk80d:
+.. _tlsr9518adk80d:
```
Comment (mbolivar-nordic) [inline boards/riscv/tlsr9518adk80d/doc/index.rst:1]: This should be part of its own commit.

File: drivers/pwm/pwm_b91.c [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+/* PWM driver APIs structure */
+static const struct pwm_driver_api pwm_b91_driver_api = {
+	.pin_set = pwm_b91_pin_set,
+	.get_cycles_per_sec = pwm_b91_get_cycles_per_sec,
+};
+
+/* PWM driver registration */
+DEVICE_DT_INST_DEFINE(0, pwm_b91_init,
```
Comment (mbolivar-nordic) [inline drivers/pwm/pwm_b91.c:132]: - Singleton drivers often have a BUILD_ASSERT which future-proofs them if multiple instances are added in a future hardware variant. See e.g. https://github.com/zephyrproject-rtos/zephyr/blob/4cb0020614f4494ce2c7779c0ccd1b1777e00727/drivers/sensor/nxp_kinetis_temp/temp_kinetis.c#L181
- I didn't request this change in the SoC flash driver because that peripheral tends to be a special case, but a PWM driver like this should really respect `status = "disabled";` in the devicetree by using DT_INST_FOREACH_STATUS_OKAY instead. See the same temp_kinetis.c driver for a code example.

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
Just a few comments from me on the DT side.

File: dts/bindings/pwm/telink,b91-pwm.yaml (hunk 1 above)
Comment (yurvyn) [reply dts/bindings/pwm/telink,b91-pwm.yaml:22]: Done.
Reworked to use booleans.

File: boards/riscv/tlsr9518adk80d/doc/index.rst (hunk 2 above)
Comment (yurvyn) [reply boards/riscv/tlsr9518adk80d/doc/index.rst:1]: Done.

File: drivers/pwm/pwm_b91.c (hunk 3 above)
Comment (yurvyn) [reply drivers/pwm/pwm_b91.c]: Done.

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by yurvyn** [comment]
One more approval is required.
Could anyone please approve the PR?

**Review by anangl — APPROVED** [review]
LGTM

**Comment by carlescufi** [comment]
@yurvyn please rebase

**Comment by yurvyn** [comment]
Rebase is done. Please merge.

**Review by carlescufi — APPROVED** [review]
